File: StopWatch_oc/StopWatch_oc/recordTimeCell.m
//
//  recordTimeCell.m
//  StopWatch_oc
//
//  Created by ynfMac on 2020/5/25.
//  Copyright © 2020 ynfMac. All rights reserved.
//

#import "recordTimeCell.h"
#import "StopWatchData.h"

@implementation recordTimeCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    
    if (self) {
        
        self.backgroundColor = [UIColor blackColor];
        self.textLabel.textColor = [UIColor whiteColor];
        self.detailTextLabel.textColor = [UIColor whiteColor];
        self.detailTextLabel.font = [UIFont fontWithName:@"Hiragino Sans W3" size:17.0f];
        self.userInteractionEnabled = NO;
    }
    
    return self;
}

- (void)recordTimeWithData:(StopWatchData *)data atIndexPath:(NSIndexPath *)indexPath{
    
    self.textLabel.textColor = [UIColor whiteColor];
    self.detailTextLabel.textColor = [UIColor whiteColor];

    self.textLabel.text = [NSString stringWithFormat:@"计次 %lu", data.times.count - indexPath.row];

    NSUInteger time = [data.times[indexPath.row] integerValue];

    self.detailTextLabel.text = [self _convertTime:time];
    self.detailTextLabel.textAlignment = NSTextAlignmentLeft;
    
    
    
    if (indexPath.row == data.maxTimeIndex) {
        self.textLabel.textColor = [UIColor redColor];
        self.detailTextLabel.textColor = [UIColor redColor];
    }

    if (indexPath.row == data.minTimeIndex) {
        self.textLabel.textColor = [UIColor greenColor];
        self.detailTextLabel.textColor = [UIColor greenColor];
    }
    
}

- (NSString *)_convertTime:(NSUInteger)time{
    return [NSString stringWithFormat:@"%02ld:%02ld.%02ld",time / 100 / 60, time / 100 % 60, time % 100];;
}



@end

File: stopWatch_swift/stopWatch_swift/RecordTimeCell.swift
//
//  RecordTimeCell.swift
//  stopWatch_swift
//
//  Created by ynfMac on 2020/5/25.
//  Copyright © 2020 ynfMac. All rights reserved.
//

import UIKit

class RecordTimeCell: UITableViewCell {

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        backgroundColor = UIColor.black
        textLabel?.textColor = UIColor.white
        detailTextLabel?.textColor = UIColor.white
        detailTextLabel?.font = UIFont.init(name: "HiraMinProN-W3", size: 17)
        isUserInteractionEnabled = false
    }
    
    public func recordTime( with data:stopWatchData, _ indexPath:IndexPath){
        
        textLabel?.textColor = UIColor.white
        detailTextLabel?.textColor = UIColor.white
        
        textLabel?.text = "计次\(data.times.count - indexPath.row)"
        let time = data.times[indexPath.row]
        detailTextLabel?.text = time.convertToTimeFormat()
        
        if indexPath.row == data.maxTime{
            textLabel?.textColor = UIColor.red
            detailTextLabel?.textColor = UIColor.red
        }
        
        if indexPath.row == data.minTime{
            textLabel?.textColor = UIColor.green
            detailTextLabel?.textColor = UIColor.green
        }
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    


}
